avm/res/document-db/database-account/main.bicep

```diff
@@ -30,26 +30,32 @@ param locations array
   'BoundedStaleness'
   'Strong'
 ])
-@description('Optional. The default consistency level of the Cosmos DB account.')
+@description('Optional. Default to Session. The default consistency level of the Cosmos DB account.')
```

Comment:
```suggestion
@description('Optional. The default consistency level of the Cosmos DB account.')
```

The default is automatically documented. Adding it also to the description means to start maintaining 2 locations instead of one. Please also update for the other parameters.

Reply:
Where is it documented? In the auto generated read me file?
Dont you think it is a better consumer experience to see it while you are writing code rather than having to jump to the documentation? In the end the readme is generated automatically and we are forcing people to regenerate it. So I would rather say that we have just one place to maintain.

In my opinion. I'm more on the side of forcing the developer to jump to the documentation as minimum as possible. They might not even know about it. Look at me. This is my first time consuming this module and I didn't even know the defaults were there and the first place i looked up was in the intelisense.

If you still consider not to have it in the description then I will make the change but I unfortunately dont agree with it

Reply:
Second thoughts about this. I have one use case where I don't have the real default value used in the parameter so it is really not documented in the Readme.md file as it is not picked up by the generation. It is not specified in the parameters as it created the bug #1226 the default value needs to be calculated on the fly

Reply:
Hey @bryansan-msft,
I don't think it's documented anywhere as it is no spec - I guess it's more of a recommendation to keep the maintenance effort at bay.

To your question you can find the metadata actually in a few locations. One is the readme, yes - but as you said, you don't want to jump back and forth. That said, the Bicep Linter got you covered too like for example this:

![image](https://github.com/Azure/bicep-registry-modules/assets/5365358/71497265-116b-43fe-af65-7ee1dd949f16)

with the 'allowed values'.

Contrary to my expectations, you're right with the defaults. I don't know why - but they are not shown. Like here where it should say `true`:
![image](https://github.com/Azure/bicep-registry-modules/assets/5365358/2498b0bc-2fbd-4183-bb34-43ff5ea0845d)

I agree with your observation, but, would actually want to see this in linter more than seeing it in the description. I'll check if I can find an issue for that. Afterall - all the other stuff is shown, why not the default?

cc: @eriqua, @ChrisSidebotham

Reply:
Update: I wasn't able to find an issue (suprisingly) and I'll open one

Reply:
Descriptions for input parameters are not that strictly limited currently. For example, we have allowed values ported there in some cases, while that is already available as Alex brought up above.
That said, I agree with @bryansan-msft on it would be better to show default values via intellisense and I agree with @AlexanderSehr on this is more a feature request for linter than it is for the avm description metadata as this may be challenging to maintain overtime and also to require consistently for all applicable input parameter descriptions. 
I'd suggest to first understand how long it would take realistically to implement that feature. Based on that we may want to provide guidelines to contributors on not to waste effort on explicitly call out allowed or default values in description metadata.

Reply:
Well, I agree that if the linter does it it would be better for everyone. But until it does it what do we do? I am keen on leaving them on the description but I am not the one deciding.

Reply:
One thing to add is that if someone does this then the default value would be missing completely from everywhere as the auto generated Readme will not pick it up

![image](https://github.com/Azure/bicep-registry-modules/assets/117635118/10244acd-782c-4b7a-8589-1e1fd776dcbf)

Reply:
correct @bryansan-msft . In somecases it's implemented like this because UDTs don't support defaults (e.g. in the diagnostic settings), in others because other logic is required. It's unfortunate, but I wouldn't know how to solve this in a good way.

Coming back to the original issue - I won't block adding the defaults to the description. Ultimately, it's your choice and if you feel it's worth-while, so be it :)

Resolving the comment.